samples/algorithms/sudoku-grover/Sudoku.qs

```diff
@@ -0,0 +1,203 @@
+    /// The constraints on these empty squares to the starting numbers are
+    /// startingNumberConstraints = (0,1)  (0,3)  (1,1)  (1,3)   
+    /// This is a list of (empty square #, number it can't be)  
+    /// i.e. empty square 0 can't have value 1 or 3, 
+    /// and empty square #1 can't have values 1 or 3
+    ///
+    ///
+    /// # Input
+    /// ## numVertices
+    /// number of blank squares
+    /// ## size
+    /// The size of the puzzle. 4 for 4x4 grid, 9 for 9x9 grid
+    /// ## emptySquareEdges
+    /// The traditional edges passed to the graph coloring algorithm which, 
+    /// in our case, are empty puzzle squares.
+    /// These edges define any "same row", "same column", "same sub-grid" 
+    /// relationships between empty cells
+    /// Look at the README.md sample output to see examples of what this is 
+    /// for different sample puzzles
+    /// ## startingNumberConstraints
+    /// The constraints on the empty squares due to numbers already in the 
+    /// puzzle when we start.
+    /// Look at the README.md sample output to see examples of what this is 
+    /// for different sample puzzles
+    ///
+    ///
+    /// # Output
+    /// A Tuple with Result and the array of numbers for each empty square
+    /// Look at the README.md sample output to see examples of what this is 
+    /// for different sample puzzles
+    ///
+    /// # Remarks
+    /// The inputs and outputs for the following 4x4 puzzle are
+    ///    -----------------
+    ///    |   | 1 | 2 | 3 |         <--- empty square #0
+    ///    -----------------
+    ///    | 2 |   | 0 | 1 |         <--- empty square #1
+    ///    -----------------
+    ///    | 1 | 2 | 3 | 0 |
+    ///    -----------------
+    ///    | 3 |   | 1 | 2 |         <--- empty square #2
+    ///    -----------------
+    ///    emptySquareEdges = [(1, 0),(2, 1)]    
+    ///         empty square #0 can not have the same color/number as empty call #1 
+    ///         empty square #1 and #2 can not have the same color/number (same column)
+    ///    startingNumberConstraints = [(0, 2),(0, 1),(0, 3),(1, 1),(1, 2),(1, 0),(2, 1),(2, 2),(2, 3)]
+    ///         empty square #0 can not have values 2,1,3 because same row/column/2x2grid
+    ///         empty square #1 can not have values 1,2,0 because same row/column/2x2grid
+    ///    Results = [0,3,0] i.e. Empty Square #0 = 0, Empty Square #1 = 3, Empty Square #2 = 0
+    operation SolvePuzzle(numVertices : Int, size : Int, emptySquareEdges : (Int, Int)[], 
+        startingNumberConstraints: (Int, Int)[]) : (Bool, Int[]) {
+        // for size = 4x4 grid
+        mutable bitsPerColor = 2; 
+        mutable oracle = VertexColoringOracle(numVertices, bitsPerColor, emptySquareEdges, startingNumberConstraints, _, _);
+        if (size == 9) {
+            // if size == 9x9 grid, the solution has 9 colors so we need 
+            // 4 bits per color
+            set bitsPerColor = 4;
```

Comment:
You may be able to simplify this further by using the `?|` operator to avoid mutability in `bitsPerColor`. For instance, something of the form `let bitsPerColor = size == 9 ? 4 | 2;`.